sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -551,6 +586,548 @@ Parameters:
+in all registered graph nodes.
+* `newAccessRange` - Access range for the accessor of the new buffer parameter.
+* `newAccessOffset` - Access offset for the range of the new buffer parameter.
+
+Exceptions:
+
+Throws an exception with the `errc::invalid` error code if the sum of
+`newAccessRange` and `newAccessOffset` exceeds the range of `bufferRef` in any dimension.
+
+|
+[source,c++]
+----
+accessor<DataT, Dimensions, AccessMode, AccessTarget> get(handler &cgh);
+----
+
+| Returns a `sycl::accessor` to use in the current command-group.
+The accessor is registered with this `dynamic_accessor` instance
+and will be updated whenever `dynamic_accessor::update()` is called.
+
+Parameters:
+
+* `handler` - The kernel handler that represents the current submission.
+
+Returns: 
+
+An instance of `sycl::accessor` to use in the current command-group.
+
+|===
+
+===== The dynamic_local_accessor class [[dynamic-local-accessor-class]]
+
+[source,c++]
+----
+namespace ext::oneapi::experimental{
+template <typename DataT, int Dimensions = 1>
+class dynamic_local_accessor {
+public:
+  dynamic_local_accessor(
+        command_graph<graph_state::modifiable> graph,
+        range<Dimensions> allocationSize,
+        const property_list &propList = {});
+
+  void update(range<Dimensions> newAllocationSize);
+
+  local_accessor<DataT, Dimensions> get(handler &cgh);
+};
+}
+----
+
+The `dynamic_local_accessor` class provides the {crs}[common reference semantics].
+
+Table {counter: tableNumber}. Member functions of the `dynamic_local_accessor` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+template <typename DataT, int Dimensions = 1>
+dynamic_local_accessor(
+        command_graph<graph_state::modifiable> graph,
+        range<Dimensions> allocationSize);
+----
+|Available only when `(Dimensions > 0)``.
+
+Constructs a dynamic local accessor object that can be registered with command graph.
+
+Parameters:
+
+* `graph` - Graph which will contain the nodes that use the dynamic local accessor.
+* `allocationSize` - The size of the local accessor.
+
+|
+[source,c++]
+----
+void update(range<Dimensions> newAllocationSize);
+----
+
+|Updates the allocation size of this dynamic local accessor in all graph nodes
+where it is registered. This new size will be reflected immediately in the
+modifiable graph which contains the registered nodes. The new size will not be
+reflected in any executable graphs created from that modifiable graph until
+`command_graph::update()` is called passing the modified nodes, or a new
+executable graph is finalized from the modifiable graph.
+
+It is not an error if `newAllocationSize` is set to the current allocation size
+of any registered nodes.
+
+Parameters:
+
+* `newAllocationSize` - The new allocation size that this dynamic local accessor will
+use in all the graph nodes where it is registered.
+
+|
+[source,c++]
+----
+local_accessor<DataT, Dimensions> get(handler &cgh);
```

Comment:
what happens if you call `get()` on a handler without doing `set_arg()`? Do we need an exception for that (which could be relaxed later once we have compiler support)

Reply:
I think calling `get()` without `set_arg()` should just give you a regular `local_accessor` that doesn't get updated on a call to update()

I think this would be more consistent with the current pattern we have with `dynamic_parameters` where `set_args is cannot be enforced. 

Adding an exception is not hard though if we think that is better. A conseguence of calling an exception is that `get()` should always be called afer `set_arg()` but that's probably not a big deal.